src/Controller/Component/RequestHandlerComponent.php

```diff
@@ -227,16 +216,6 @@ public function startup(Event $event)
         }
 
         $request->params['isAjax'] = $this->request->is('ajax');
-        $isRecognized = (
-            !in_array($this->ext, ['html', 'htm']) &&
-            $this->response->getMimeType($this->ext)
-        );
-
-        if (!empty($this->ext) && $isRecognized) {
-            $this->renderAs($controller, $this->ext);
-        } elseif (empty($this->ext) || in_array($this->ext, ['html', 'htm'])) {
-            $this->respondAs('html', ['charset' => Configure::read('App.encoding')]);
-        }
```

Comment:
This was always rendering logic anyways. It makes more sense in beforeRender. However, its something we should probably mention in the upgrade guide for 3.1.